### PR TITLE
R data preparation to add age rescaling and determinism

### LINKED ISSUE #63: Make first step (raw_to_prepared.R) fully runnable
Currently, after running `raw_to_prepared.R`, the four `ageRescale---.csv` files are missing from [the zip file](https://github.com/alan-turing-institute/uatk-spc/blob/main/scripts/data_prep/SAVE_SPC_required_data.zip).

To restore them, it is necessary to:

### 3. Run some bits of `SPC_pipelineLAD.R` for all of England
https://github.com/alan-turing-institute/uatk-spc/blob/main/scripts/data_prep/SPC_pipelineLAD.R#L1-L58
and
https://github.com/alan-turing-institute/uatk-spc/blob/main/scripts/data_prep/SPC_pipelineLAD.R#L92-L140

Requires:
- Content of step 1
- [The zip file](https://github.com/alan-turing-institute/uatk-spc/blob/main/scripts/data_prep/SAVE_SPC_required_data.zip) created during step 2
-  l. 134: `addToData2` and it's dependencies (loaded from https://github.com/alan-turing-institute/uatk-spc/blob/main/scripts/data_prep/SPC_functions.R#192-L331) must be replaced by `addToData` (and its dependencies) from
https://github.com/alan-turing-institute/uatk-spc/blob/main/scripts/data_prep/raw_to_prepared_Income.R#L216-314,
after adaptation (time at work `(data$pwork[idx]+data$pworkhome[idx])` l. 262-269-272 should be taken from `drawTime` instead of the data since it no longer exists there. Maybe other adaptations?)

### 4. Run the rest of `raw_to_prepared_Income.R`

First, load the data created during step 3 following:
https://github.com/alan-turing-institute/uatk-spc/blob/main/scripts/data_prep/raw_to_prepared_Income.R#L326-341
And use it to run:
https://github.com/alan-turing-institute/uatk-spc/blob/main/scripts/data_prep/raw_to_prepared_Income.R#L583-726

This should create the missing files that can then be added to [the zip file](https://github.com/alan-turing-institute/uatk-spc/blob/main/scripts/data_prep/SAVE_SPC_required_data.zip).

--

Then, to get the Azure data:

### 5. Run the current version of `SPC_pipelineLAD.R`

```
source("SPC_loadWorkspace.R")

# Select year
date <- 2012

# Select lad
lad <- "E06000002"
lads <- sort(unique(lu$LAD20CD[lu$Country == "England"]))
  
for(i in lads){
  lad <- lads[i]
  source("SPC_pipelineLAD.R")
}
```

### PR DESCRIPTION
**Pull request by sgreenbury**
Closes #63.

### CODE REVIEW

**Comment by sgreenbury** [comment]
Remaining tasks:
- [ ] Confirm determinism for [step 2](https://github.com/alan-turing-institute/uatk-spc/tree/main/scripts/data_prep#step-2-add-to-spenser)
- [ ] Add script to move [step 1 data](https://github.com/alan-turing-institute/uatk-spc/tree/main/scripts/data_prep#step-1-curate-public-data-from-diverse-sources) files to expected input path `scripts/data_prep/Data/dl/`
- [ ] Add manifest for [step 1 data](https://github.com/alan-turing-institute/uatk-spc/tree/main/scripts/data_prep#step-1-curate-public-data-from-diverse-sources)

File: scripts/data_prep/age_rescaling/SPC_loadWorkspace_age_rescaling.R [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+source("age_rescaling/SPC_functions_age_rescaling.R")
+
+# Lookup
+lu <- read.csv(paste(folderInOT, "lookUp-GB.csv", sep = ""))
+
+# Income related data // Only necessary for England
+# ageRescaleFFT <- read.csv(paste(folderInOT,"ageRescaleFFT.csv", sep = ""))
+# ageRescaleFPT <- read.csv(paste(folderInOT,"ageRescaleFPT.csv", sep = ""))
+# ageRescaleMFT <- read.csv(paste(folderInOT,"ageRescaleMFT.csv", sep = ""))
+# ageRescaleMPT <- read.csv(paste(folderInOT,"ageRescaleMPT.csv", sep = ""))
```
Comment (sgreenbury) [inline scripts/data_prep/age_rescaling/SPC_loadWorkspace_age_rescaling.R:19]:
```suggestion
```

**Comment by sgreenbury** [comment]
Upon re-running steps 1 (all outputs in `Data/prepData`) and step 2 (tested for E09000001 and E09000001), the outputs are now deterministically produced from the given random seed on a given machine.

**Review by HSalat — APPROVED** [review]
Quite interesting! I should learn how to do all this.

**Comment by sgreenbury** [comment]
> Quite interesting! I should learn how to do all this.

Thanks for taking a look and the review!

**Pull request merged**